api/tests/unit/segments/test_unit_segments_models.py

```diff
@@ -1,28 +1,44 @@
+from collections.abc import Callable
+from typing import Any
 from unittest.mock import PropertyMock
 
-from flag_engine.segments.constants import EQUAL, PERCENTAGE_SPLIT
+import pytest
+from django.core.exceptions import ValidationError
+from flag_engine.segments.constants import EQUAL
 from pytest_mock import MockerFixture
 
 from segments.models import Condition, Segment, SegmentRule
-from segments.services import SegmentCloneService
 
 
 def test_get_segment_returns_parent_segment_for_nested_rule(
     segment: Segment,
 ) -> None:
     # Given
-    parent_rule = SegmentRule.objects.create(segment=segment, type=SegmentRule.ALL_RULE)
-    child_rule = SegmentRule.objects.create(rule=parent_rule, type=SegmentRule.ALL_RULE)
-    grandchild_rule = SegmentRule.objects.create(
-        rule=child_rule, type=SegmentRule.ALL_RULE
+    rule1 = SegmentRule.objects.create(segment=segment, type=SegmentRule.ALL_RULE)
+    rule2 = SegmentRule.objects.create(rule=rule1, type=SegmentRule.ALL_RULE)
+    rule3 = SegmentRule.objects.create(rule=rule2, type=SegmentRule.ALL_RULE)
```

Comment:
It's interesting because I believe we allow this case while we don't have any way to deep-nest rules above 1 level right?
What would happen ? 
```
        source_rules = SegmentRule.objects.filter(
            models.Q(segment=source_segment) | models.Q(rule__segment=source_segment)
        )
```
I don't think it would pick it up would it ?

Maybe worth adding a test of the expected behavior (be it infinite recursive copy or 2levels) so that we have a source of information later on ?

Reply:
I agree this test case does not represent a real scenario. Right now the segment serializer will only allow nesting rules one level deep, e.g. `{rules: [{rules: [{conditions: ...}, conditions: ...]}`. I've made a small improvement to this test (80712382d). If we have time today I'd love to also add a test to ensure `clone` will ignore/error deeper rules but I can also follow up on #5825 if that works for you. IMO I think we're safe.